sensor/common/scannerclient/grpc_client.go

```diff
@@ -164,7 +164,7 @@ func dialV2() (ScannerClient, error) {
 // dialV4 connect to scanner V4 gRPC and return a new ScannerClient.
 func dialV4() (ScannerClient, error) {
 	ctx := context.Background()
-	c, err := client.NewGRPCScanner(ctx, client.WithIndexerAddress(env.ScannerV4IndexerEndpoint.Setting()))
+	c, err := client.NewGRPCScanner(ctx, client.IndexerOnly(), client.WithIndexerAddress(env.ScannerV4IndexerEndpoint.Setting()))
```

Comment:
Why don't we use `WithAddress(address)` instead of `client.WithIndexerAddress(address)`?

Reply:
For this particular use case, it doesn't really matter. Two reasons why I went it though:

1. It was already there, so the changes are minimized
2. Makes it clear our main focus here is the Indexer and not also the Matcher

Reply:
This was changed from `WithAddress` at #9555 for no apparent reason. It looks like it was a misunderstanding of the client's design. The options for setting addresses are not related to the mode of the Scanner we're connecting to. The client was meant to connect to either one instance (combo) or two separate instances (split). If an API is not supported, the server should return "UnsupportedOperation.".

What was the reason for the move to `WithIndexerAddress()`? If none, let's go back to `WithAddress()`.

Reply:
This was changed from `WithAddress` at #9555 for no apparent reason. It looks like it was a misunderstanding of the client's design. The options for setting addresses are not related to the mode of the Scanner we're connecting to. The client was meant to connect to either one instance (combo) or two separate instances (split). If an API is not supported, the server should return "UnsupportedOperation.".

What was the reason for the move to `WithIndexerAddress()`? If none, let's go back to `WithAddress()`.

Reply:
[2da93e6](https://github.com/stackrox/stackrox/pull/11161/commits/2da93e6646395df9c3e4cf2f5b69066b161503e6)